src/connection.js

```diff
@@ -1696,6 +1808,47 @@ Connection.prototype.STATE = {
       }
     }
   },
+  SENT_LOGIN7_WITH_FEDAUTH: {
+    name: 'SentLogin7Withfedauth',
```

Comment:
Can you split this into 2 states, as per `3.2.1 Abstract Data Model` and name them to be inline with  `Sent LOGIN7 Record with Complete Authentication Token State`(3.2.5.3) and `Sent LOGIN7 Record with Federated Authentication Information Request State`(3.2.5.5)?